### ISSUE TITLE
Cosmos: Allow users to pass in a QueryRequestOptions on a per-query basis

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#34096]
Cosmos allows tweaking query execution in many ways via [QueryRequestOptions](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.queryrequestoptions?view=azure-dotnet); we could have a WithQueryRequestOptions() operator to allow the user to specify this.

Note that we already set PartitionKey and MaxItemCount.

**Comment by roji** [comment]
Consider the behavior when EF wants to set a request option and the user has passed in the object - does EF override the user option, etc.

For now, we'll consider this as a stretch goal for EF 10.

**Comment by Pilchie** [comment]
Is there a way to pass a request options to a lambda to allow the user to modify it? Of course, we have to decide on ordering and who gets the last word...

**Comment by roji** [comment]
@Pilchie not currently - that's what this issue tracks. I'm currently thinking of a per-query mechanism such as the following:

```c#
var stuff = context.Blogs
    .Where(b => b.Name.StartsWith("X"))
    .WithRequestOptions(new() { /* the stuff */ })
    .ToListAsync();
```

Does that seem ok?

**Comment by roji** [comment]
@Pilchie makes sense - yeah, it depends on the who-wins question. We'll take all this into account when designing, thanks!
